docs/src/background/ode_solvers.md

```diff
@@ -0,0 +1,267 @@
+```
+
+Since solving a system of ``s`` coupled equations for the stages ``U_i`` is usually impractical, many of the ``a_{i,j}`` coefficients are set to 0 in commonly used Runge-Kutta methods. When ``a_{i,j} = 0`` for all ``j \geq i``, the equation for ``U_i`` simplifies to the explicit formula
+
+```math
+U_i = u_0 + \Delta t \sum_{j = 1}^{i - 1} a_{i,j} T(U_j, t_0 + \Delta t c_j).
+```
+
+This is called an explicit Runge-Kutta (ERK) method. When ``a_{i,j} = 0`` for all ``j > i``, the implicit equation for ``U_i`` becomes uncoupled from the equations for the other stages, and it can be rewritten as
+
+```math
+u_0 + \Delta t \sum_{j = 1}^{i - 1} a_{i,j} T(U_j, t_0 + \Delta t c_j) + \Delta t a_{i,i} T(U_i, t_0 + \Delta t c_i) - U_i = 0.
+```
+
+Since the only unknown tendency in this equation comes from the diagonal coefficient ``a_{i,i}``, this is called a *diagonally implicit* Runge-Kutta (DIRK) method. There are many different categories of DIRK methods, such as *explicit first stage* DIRK (EDIRK), where ``a_{1,1} = 0``, *singly* DIRK (SDIRK), where ``a_{1,1} = a_{2,2} = \ldots = a_{s,s} = \gamma`` for some constant ``\gamma``, and ESDIRK, where ``a_{1,1} = 0`` and ``a_{2,2} = \ldots = a_{s,s} = \gamma``.
+
+Just as the forward and backward Euler methods can be combined into the IMEX Euler method, two Runge-Kutta methods can be combined into an *additive* Runge-Kutta (ARK) method. If the first method is explicit and the second is implicit, the result is an IMEX ARK method. We will only be considering IMEX ARK methods where the implicit part is DIRK. If the DIRK method has coefficients ``a_{i,j}``, ``b_i``, and ``c_i``, and the ERK method has coefficients ``\tilde{a}_{i,j}``, ``\tilde{b}_i``, and ``\tilde{c}_i``, the IMEX ARK method approximates ``u(t_0 + \Delta t)`` as
+
+```math
+\hat{u} = u_0 + \Delta t \sum_{i = 1}^s \left(\tilde{b}_i T_{\text{exp}}(U_i, t_0 + \Delta t \tilde{c}_i) + b_i T_{\text{imp}}(U_i, t_0 + \Delta t c_i)\right),
+```
+
+where the implicit equation for ``U_i`` is now
+
+```math
+u_0 + \Delta t \sum_{j = 1}^{i - 1} \left(\tilde{a}_{i,j} T_{\text{exp}}(U_j, t_0 + \Delta t \tilde{c}_j) + a_{i,j} T_{\text{imp}}(U_j, t_0 + \Delta t c_j)\right) + \Delta t a_{i,i} T_{\text{imp}}(U_i, t_0 + \Delta t c_i) - U_i = 0.
+```
+
+## Adding DSS
+
+It is often necessary to filter the state so that it satisfies some particular constraint before it is used to evaluate a tendency. In our case, the state is a collection of values defined across a spatially discretized domain. When we use a discontinuous Galerkin spectral element discretization, we must ensure that the state is continuous across element boundaries before we use it to compute any tendency. We can turn any discontinuous state into a continuous one by applying a *direct stiffness summation* (DSS) to it. Applying DSS to ``\hat{u}`` in the IMEX ARK method is straightforward:
```

Comment:
```suggestion
It is often necessary to filter the state so that it satisfies some particular constraint before it is used to evaluate a tendency. In our case, the state is a collection of values defined across a spatially discretized domain. When we use a continuous Galerkin (CG) spectral element discretization, we must ensure that the state is continuous across element boundaries before we use it to compute any tendency. We can turn any state that is discontinuous across element boundaries into a continuous one by applying a *direct stiffness summation* (DSS) to it. Applying DSS to ``\hat{u}`` in the IMEX ARK method is straightforward:
```